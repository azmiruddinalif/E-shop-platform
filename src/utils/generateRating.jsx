import { FaRegStar, FaStar } from "react-icons/fa";
import { FaRegStarHalfStroke } from "react-icons/fa6";

export const getStars = (rating) => {
  const fullStars = Math.floor(rating);
  const halfStars = rating % 1 >= 0.25 && rating % 1 < 0.75;
  const emptyStars = 5 - fullStars - (halfStars ? 1 : 0);

  const stars = [];

  for (let i = 0; i < fullStars; i++) {
    stars.push(
      <span key={`full-${i}`} className="text-yellow">
        <FaStar />
      </span>
    );
  }

  if (halfStars) {
    stars.push(
      <span key={`half`} className="text-yellow">
        <FaRegStarHalfStroke />
      </span>
    );
  }

  for (let i = 0; i < emptyStars; i++) {
    stars.push(
      <span key={`empty-${i}`} className="text-yellow">
        <FaRegStar />
      </span>
    );
  }
  return <div className="flex items-center gap-x-1">{stars}</div>;
};
