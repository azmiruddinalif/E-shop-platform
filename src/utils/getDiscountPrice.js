export const getDiscountedPrice = (price, discount) => {
  if (!discount || discount <= 0) return price;
  const discountedPrice = price - (price * discount) / 100;
  return parseFloat(discountedPrice.toFixed(2));
};
