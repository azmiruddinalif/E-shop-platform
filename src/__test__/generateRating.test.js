import { expect, it } from "vitest";
import { getStars } from "../utils/generateRating";
import { getDiscountedPrice } from "../utils/getDiscountPrice";

it("should getting rate of users", () => {
  const rating = getStars(4.5);
  expect(rating).toMatchSnapshot();
});

it("should be give a discounted price", () => {
  const discountPrice = getDiscountedPrice(500, 10);
  expect(discountPrice).toBe(450);
});
