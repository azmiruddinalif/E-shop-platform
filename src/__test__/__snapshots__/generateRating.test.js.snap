// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should getting rate of users 1`] = `
<div
  className="flex items-center gap-x-1"
>
  <span
    className="text-yellow"
  >
    <FaStar />
  </span>
  <span
    className="text-yellow"
  >
    <FaStar />
  </span>
  <span
    className="text-yellow"
  >
    <FaStar />
  </span>
  <span
    className="text-yellow"
  >
    <FaStar />
  </span>
  <span
    className="text-yellow"
  >
    <FaRegStarHalfStroke />
  </span>
</div>
`;
